src/tool_paramhlp.c

```diff
@@ -83,6 +83,8 @@ ParameterError file2string(char **bufp, FILE *file)
       stringlen += buflen;
     }
   }
+  if(*bufp)
+    free(*bufp);
```

Comment:
Is there a reason to defer this rather than doing it immediately on entering the function? Careless use of the function which isn't checking the returnvalue might perhaps otherwise be tricked into thinking `bufp` has new data when in fact the `realloc()` call failed. Under extreme memory pressure it might also be beneficial to `free` before `realloc`, but that's a secondary concern.

Reply:
You're right, it should rather do this at entry and leave it cleared if it later fails...